shm/cargo.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <string>
+
+class Cargo
+{
+public:
+    Cargo &operator+=(size_t amount);
+    Cargo &operator-=(size_t amount);
+    bool operator==(const Cargo &cargo) const;
+
+    std::string getName() { return name_; }
```

Comment:
You can think about time of living `Cargo` object and usage of `getName` function, maybe you can avoid copy string here and return a const reference